packages/components/src/components/hds/side-nav/index.ts

```diff
@@ -9,20 +9,63 @@ import { action } from '@ember/object';
 import { assert } from '@ember/debug';
 import { registerDestructor } from '@ember/destroyable';
 
-export default class HdsSideNavComponent extends Component {
+import type { HdsSideNavBaseSignature } from './base';
+
+interface HdsSideNavSignature {
+  Args: {
+    isResponsive?: boolean;
+    isCollapsible?: boolean;
+    isMinimized?: boolean;
+    hasA11yRefocus?: boolean;
+    a11yRefocusSkipTo?: string;
+    a11yRefocusSkipText?: string;
+    a11yRefocusNavigationText?: string;
+    a11yRefocusRouteChangeValidator?: string;
+    ariaLabel?: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onToggleMinimizedStatus?: (arg: boolean) => void;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onDesktopViewportChange?: (arg: boolean) => void;
+  };
+  Blocks: {
+    header?: [
+      {
+        Header?: HdsSideNavBaseSignature['Blocks']['header'];
+        isMinimized?: boolean;
+      }
+    ];
+    body?: [
+      {
+        Body?: HdsSideNavBaseSignature['Blocks']['body'];
+        isMinimized?: boolean;
+      }
+    ];
+    footer?: [
+      {
+        Footer?: HdsSideNavBaseSignature['Blocks']['footer'];
+        isMinimized?: boolean;
+      }
+    ];
+  };
+  Element: HdsSideNavBaseSignature['Element'];
```

Comment:
question: is there documentation somewhere that demonstrates how/why/when to do this kind of element definition specifically? I'm asking because I thought that this was constrained to the existing/valid HTML Element list, so I'm curious to learn more.

Reply:
@alex-ju can you keep me honest on the above? 👆

Reply:
what Cristiano said is accurate; to help cascading/propagating types we oftentimes use the signature of the referenced component